pkg/slicegwrecycler/recycler.go

```diff
@@ -0,0 +1,207 @@
+		// post event to slicegw
+		utils.RecordEvent(ctx, eventRecorder, sliceGw, slice, ossEvents.EventSliceGWRemotePodSyncFailed, controllerName)
+		return false, err
+	}
+	err = hubClient.CreateWorkerSliceGwRecycler(ctx, gwRecyclerName, clientID, gatewayPod.Name, sliceGw.Name, sliceGw.Status.Config.SliceGatewayRemoteGatewayID, sliceGw.Spec.SliceName)
+	if err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	utils.RecordEvent(ctx, eventRecorder, sliceGw, slice, ossEvents.EventSliceGWRebalancingSuccess, controllerName)
+	// spawn a new gw nodeport service
+	_, _, _, err = handleSliceGwSvcCreation(ctx, meshClient, hubClient, sliceGw, NumberOfGateways+1, eventRecorder, controllerName)
+	if err != nil {
+		//TODO:add an event and log
+		return false, err
+	}
+	return true, nil
+}
+
+// The function was relocated to this location in order to consolidate it into a central location
+// and facilitate its use as a utility function.
+func handleSliceGwSvcCreation(ctx context.Context, meshClient client.Client, hubClient *hub.HubClientConfig, sliceGw *kubeslicev1beta1.SliceGateway, n int, eventRecorder *monitoringEvents.EventRecorder, controllerName string) (bool, reconcile.Result, []int, error) {
+	log := logger.FromContext(ctx).WithName("slicegw")
+	sliceGwName := sliceGw.Name
+	foundsvc := &corev1.Service{}
+	var sliceGwNodePorts []int
+	no, _ := getNumberOfGatewayNodePortServices(ctx, meshClient, sliceGw)
+	if no != n {
+		// capping the number of services to be 2 for now, later i will be equal to number of gateway nodes,eg: i = len(node.GetExternalIpList())
+		for i := 0; i < n; i++ {
+			err := meshClient.Get(ctx, types.NamespacedName{Name: "svc-" + sliceGwName + "-" + fmt.Sprint(i), Namespace: controllers.ControlPlaneNamespace}, foundsvc)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					svc := serviceForGateway(sliceGw, i)
+					log.Info("Creating a new Service", "Namespace", svc.Namespace, "Name", svc.Name)
+					err = meshClient.Create(ctx, svc)
+					if err != nil {
+						log.Error(err, "Failed to create new Service", "Namespace", svc.Namespace, "Name", svc.Name)
+						return true, ctrl.Result{}, nil, err
+					}
+					return true, ctrl.Result{Requeue: true}, nil, nil
+				}
+				log.Error(err, "Failed to get Service")
+				return true, ctrl.Result{}, nil, err
+			}
+		}
+	}
+	sliceGwNodePorts, _ = getNodePorts(ctx, meshClient, sliceGw)
+	err := hubClient.UpdateNodePortForSliceGwServer(ctx, sliceGwNodePorts, sliceGwName)
+	if err != nil {
+		log.Error(err, "Failed to update NodePort for sliceGw in the hub")
+		utils.RecordEvent(ctx, eventRecorder, sliceGw, nil, ossEvents.EventSliceGWNodePortUpdateFailed, controllerName)
+		return true, ctrl.Result{}, sliceGwNodePorts, err
+	}
+	return false, reconcile.Result{}, sliceGwNodePorts, nil
+}
+
+func createHeadlessServiceForGwServer(slicegateway *kubeslicev1beta1.SliceGateway) *corev1.Service {
```

Comment:
Don't see this function called any where? Why do we need this API?
Can you please description for the function